packages/syft/src/syft/service/action/action_object.py

```diff
@@ -881,8 +942,8 @@ def __post_init__(self) -> None:
             if side_effect not in self._syft_post_hooks__[HOOK_ALWAYS]:
                 self._syft_post_hooks__[HOOK_ALWAYS].append(side_effect)
 
-        if isinstance(self.syft_action_data, ActionObject):
-            raise Exception("Nested ActionObjects", self.syft_action_data)
+        if isinstance(self.syft_action_data_type, ActionObject):
```

Comment:
should this be `issubclass` instead of `isinstance` since `syft_action_data_type` is a type?